consensus/types/src/chain_spec.rs

```
@@ -2031,7 +2031,8 @@ fn max_data_columns_by_root_request_common<E: EthSpec>(max_request_blocks: u64)
 
     let empty_data_columns_by_root_id = DataColumnsByRootIdentifier {
         block_root: Hash256::zero(),
-        columns: VariableList::from(vec![0; E::number_of_columns()]),
+        columns: VariableList::try_from(vec![0; E::number_of_columns()])
+            .expect("VariableList::try_from should succeed"),
```

This function could also be a Result<usize, Error>

---

This one seems like it could be a bit of an annoying one to ripple, the error would bubble up to `ChainSpec::apply_to_chain_spec` which returns an `Option` (we'd probably have to silently drop the detail of the error and return `None`).

There is also a precedent for using `expect` in this function, there's an existing one on the line below this.

I think the cleanest fix would actually be to add `VariableList::repeat_full(0)` which creates a `VariableList<T, N>` of length `N` full of the provided value. This function would not need to return a `Result`. For `FixedVector` it could be even cleaner, just `::repeat(0)`.

---

Done in 8de1f30f5d6340b382eac312fef1652744c7e7d4